### PR TITLE
react-cropper: support ref prop created with useRef hook

### PR DESCRIPTION
**Pull request by frisi**
allows to also pass refs created with the `useRef` hook. see https://github.com/roadmanfong/react-cropper/issues/134#issuecomment-593984770

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  * https://github.com/roadmanfong/react-cropper/issues/134
  * https://stackoverflow.com/questions/59437886/typescript-importing-problem-on-cropperjs-library
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by frisi** [comment]
@bwlt and @stepancar, anything else to do here or is this okay to merge?

people are having issues when using the `useRef` hook with the current cropper type definition
* https://github.com/roadmanfong/react-cropper/issues/134
* https://stackoverflow.com/questions/59437886/typescript-importing-problem-on-cropperjs-library

File: types/react-cropper/index.d.ts
Comment (bwlt) [inline types/react-cropper/index.d.ts]: @frisi I don't known if `Cropper | undefined` is accurate:
- on the right we are using `ReactCropper` instead of `Cropper`
- when using this ref I expect to have a null value or a `ReactCropper` type; undefined is not accurate here

At this point I suggest to modify the interface like this:
```
interface ReactCropperProps_ extends
  Cropper.Options, Omit<React.HTMLProps<HTMLImageElement>, 'data' | 'ref'>,
  React.RefAttributes<ReactCropper> {}
```
In this way we are reusing the `React.RefAttributes` instead of enumerating one by one like we are doing in this case.
Do you agree?

Comment (frisi) [reply types/react-cropper/index.d.ts]: sure - agreed. thanks for your reply and proposal @bwlt !

did not know about these interfaces - much cleaner this way.
however, the `LegacyRef` (string) will not be supported this way.
dunno if somebody else still needs it, but we could use that to be sure this usecase also works.

on the other side people using older react-versions that require refs as string (i guess this has something to do with the react version) can still use an older version of @types/react-cropper

``` 
    type Ref<T> = RefCallback<T> | RefObject<T> | null;
    type LegacyRef<T> = string | Ref<T>;

   interface RefAttributes<T> extends Attributes {
        ref?: Ref<T>;
    }
    interface ClassAttributes<T> extends Attributes {
        ref?: LegacyRef<T>;
    }
```

i'll update the pr and hope you can merge and release soon ;-)

Comment (bwlt) [reply types/react-cropper/index.d.ts]: Maybe it is better to maintain the retrocompatibility instead. The LegacyRef is more appropriate in this case

Comment (frisi) [reply types/react-cropper/index.d.ts]: i also chose to do still support it as there was also a tests for that.
added a test for `FunctionComponents` using `useRef` hook too.

is there anything else to do for me or can you take care of a release @bwlt ?

Comment (bwlt) [reply types/react-cropper/index.d.ts]: I'm only a reviewer :)
The ref related changes looks good to me, I don't know the impact for the other changes like this: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42801/files#diff-e7572847de7a9e112a5be7b0a58c7c31R7 .
Better to await another reviewer

Comment (frisi) [reply types/react-cropper/index.d.ts]: the update of the minimal typescript version was necessary to make tests pass and was suggested by the test output:

```
/sandbox/DefinitelyTyped/types/react-cropper/react-cropper-tests.tsx:37:37
ERROR: 37:37  expect     Compile error in typescript@3.6 but not in typescript@3.7.
Fix with a comment '// Minimum TypeScript Version: 3.7' just under the header.
Expression expected.
ERROR: 37:51  expect     Compile error in typescript@3.6 but not in typescript@3.7.
Fix with a comment '// Minimum TypeScript Version: 3.7' just under the header.
':' expected.
```

so i'll wait for another review then.... have a nice weekend!

**Comment by orta** [comment]
Cool, this looks good to me 👍

**Review by orta — APPROVED** [review]

**Comment by frisi** [comment]
thanks for merging @orta!

**Pull request merged**
